packages/node/src/client.ts

```diff
@@ -146,14 +162,78 @@ class BearerClient {
       data: payload
     })
 
-    return this.client.request<TData>({
+    const config = {
       method,
       headers,
       baseURL,
       url: endpoint,
       params: requestParams,
       data: payload
-    })
+    }
+
+    const hash = this.makeHash({ ...config, url: `${baseURL}${endpoint}` })
+    const state = this.getRequestState(hash)
+
+    state.request = () => {
+      return this.client.request<TData>(config)
+    }
+    state.numRetries = DEFAULT_NUM_RETRIES
```

Comment:
Shouldn' we use settings here?
```suggestion
    state.numRetries = this.numRetries
```